src/OpenSearch.Client/Descriptors.Indices.cs

```diff
@@ -264,6 +264,7 @@ public CreateIndexDescriptor Index<TOther>()
 			where TOther : class => Assign(typeof(TOther), (a, v) => a.RouteValues.Required("index", (IndexName)v));
 		// Request parameters
 		///<summary>Whether a type should be expected in the body of the mappings.</summary>
+		///<remarks>Deprecated as of OpenSearch 2.0</remarks>
 		public CreateIndexDescriptor IncludeTypeName(bool? includetypename = true) => Qs("include_type_name", includetypename);
```

Comment:
We should consider using the Obsolete attribute
https://stackoverflow.com/questions/314505/how-to-mark-a-class-as-deprecated

Reply:
It raises warnings, which interpreted as error according to the project settings. `[Obsolete]` attribute would fail the build on compilation stage.
Furthermore, it is not obsolete if OpenSearch 1.0 is used.

Reply:
This message goes to generated XML doc and to IDE hints:
![image](https://user-images.githubusercontent.com/88679692/177244886-bf37a31e-7293-43bf-9a99-c2a64a5186c5.png)